motionplan/solverFrame.go

```diff
@@ -210,13 +212,20 @@ func (sf *solverFrame) Interpolate(from, to []frame.Input, by float64) ([]frame.
 	}
 	interp := make([]frame.Input, 0, len(to))
 	posIdx := 0
-	for _, frame := range sf.frames {
-		dof := len(frame.DoF()) + posIdx
+	for _, currFrame := range sf.frames {
+		dof := len(currFrame.DoF()) + posIdx
 		fromSubset := from[posIdx:dof]
 		toSubset := to[posIdx:dof]
 		posIdx = dof
+		var interpSub []frame.Input
+		var err error
 
-		interpSub, err := frame.Interpolate(fromSubset, toSubset, by)
+		// if we are dealing with the execution frame, no need to interpolate, just return what we got
+		if strings.Contains(currFrame.Name(), "ExecutionFrame") {
```

Comment:
We can't do this. This package shouldn't need to know anything about the particulars of things like this.

Reply:
yea... you're right. I will need to think about this more deeply to see how we can mitigate the need for this to exist.